addons/base_import/models/base_import.py

```diff
@@ -476,107 +596,239 @@ def _try_match_date_time(self, preview_values, options):
+                Coming from "get_fields_tree" method
+                E.g: [ { 'name': 'fieldName', 'string': 'fieldLabel', fields: [ { 'name': 'subfieldName', ...} ]} , ... ]
+            :param (list) header_types: Extracted field types for each column in the parsed file, based on its data content.
+                Coming from "_extract_header_types" method
+                E.g.: ['int', 'float', 'char', 'many2one', ...]
+            :param (dict) mapping_fields: contains the previously saved mapping between header and field for the current model.
+                E.g.: { header_name: field_name }
+            :returns: if the header couldn't be matched : an empty dict
+                      else: a dict with the field path and the distance between header and the matched field.
+            :rtype: dict(field_path + Word distance)
+                    In case of simple matching : {'field_path': [field_name], distance: word_distance}
+                                           e.g.: {'field_path': ['lead_id'], distance: 0.23254}
+                    In case of hierarchy matching : {'field_path': [parent_field_name, child_field_name, subchild_field_name]}
+                                              e.g.: {'field_path': ['lead_id', 'description']}
         """
-        string_match = None
-        IrTranslation = self.env['ir.translation']
-        for field in fields:
-            # FIXME: should match all translations & original
-            # TODO: use string distance (levenshtein? hamming?)
-            if header.lower() == field['name'].lower():
-                return [field]
-            if header.lower() == field['string'].lower():
-                # matching string are not reliable way because
-                # strings have no unique constraint
-                string_match = field
-            translated_header = IrTranslation._get_source('ir.model.fields,field_description', 'model', self.env.lang, header).lower()
-            if translated_header == field['string'].lower():
-                string_match = field
-        if string_match:
-            # this behavior is only applied if there is no matching field['name']
-            return [string_match]
+        if not fields_tree:
+            return {}
+
+        # First, check in saved mapped fields
+        mapping_field_name = mapping_fields.get(header.lower())
+        if mapping_field_name and mapping_field_name:
+            return {
+                'field_path': [name for name in mapping_field_name.split('/')],
+                'distance': -1  # Trick to force to keep that match during mapping deduplication.
+            }
 
         if '/' not in header:
-            return []
+            # Then, try exact match
+            IrTranslation = self.env['ir.translation']
+            translated_header = IrTranslation._get_source('ir.model.fields,field_description', 'model', self.env.lang, header).lower()
```

Comment:
All the labels (string, help, selection labels) returned by `fields_get()` are translated.
Why going into actual implementation, which might change?